sklearn/decomposition/tests/test_fastica.py

```diff
@@ -289,7 +307,49 @@ def test_fastica_errors():
         fastica(X, algorithm="pizza")
 
 
-@pytest.mark.parametrize("whiten", [True, False])
+def test_fastica_whiten_unit_variance():
+    """Test unit variance of transformed data using FastICA algorithm.
+
+    Bug #13056
+    """
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((100, 10))
+    n_components = X.shape[1]
+    ica = FastICA(n_components=n_components, whiten="unit-variance", random_state=0)
+    Xt = ica.fit_transform(X)
+
+    assert np.var(Xt) == pytest.approx(1.0)
+
+
+@pytest.mark.parametrize("ica", [FastICA(), FastICA(whiten=True)])
+def test_fastica_whiten_default_value_deprecation(ica):
+    """Test FastICA whiten default value deprecation.
+
+    Regression test for #19490
+    """
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((100, 10))
+    with pytest.warns(FutureWarning, match=r"From version 1.3 whiten="):
+        ica.fit(X)
+        assert ica._whiten == "arbitrary-variance"
+
+
+def test_fastica_whiten_backwards_compatibility():
+    """Test previous behavior for FastICA whitening (whiten=True)
+
+    Regression test for #19490
+    """
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((100, 10))
+    n_components = X.shape[1]
+    ica = FastICA(n_components=n_components, whiten=True, random_state=0)
+    with pytest.warns(FutureWarning):
+        Xt = ica.fit_transform(X)
+
+    assert np.var(Xt) == pytest.approx(1.0 / 100)
```

Comment:
I think those last two tests could be combined and expanded to check that:

- whiten=True and default value lead to the same result as fitting `whiten='arbitrary-variance'` only using the public API (no need to introspect `ica._whiten` but does not hurt either)
- the first 2 trigger the warning and the latter does not.

Reply:
I am not sure about what you mean by "the last two tests". Are you referring to this file last test, i.e. `test_fastica_output_shape`? To me, its purposes are somewhat distinct from this test's (i.e `test_fastica_whiten_backwards_compatibility`'s).

d6c4c5e67d2a9fdcf5ec3594a2fda8eb4ec67492 is an attempt to treat your suggestion. Let me know what you think of it.

Reply:
LGTM, thanks!